### PR TITLE
make adamLoad public

### LINKED ISSUE #687: Make `adamLoad` public again
[`f9a80a4`](https://github.com/bigdatagenomics/adam/commit/f9a80a44b2142e311859254caf5eced0f6e3d4ea#diff-d36ea7d0742decd0b040a73a96af06e9R129) made `ADAMContext.adamLoad` `private[rdd]`.

I've been using `adamLoad` (circa `0.16.0`) in [pageant](https://github.com/hammerlab/pageant) to load `RDD`s of my own avro/parquet records, and just noticed I am broken by this change when building against a locally-built `0.16.1-SNAPSHOT`.

* Is it important to keep `adamLoad` package-private?
* Is there some other place that I should depend on to get this kind of parquet/avro-reading (with predicate/projection goodness) functionality?

### PR DESCRIPTION
**Pull request by ryan-williams**
fixes #687

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 1]
```diff
@@ -126,7 +126,7 @@ class ADAMContext(val sc: SparkContext) extends Serializable with Logging {
    * @tparam T The type of records to return
    * @return An RDD with records of the specified type
    */
-  private[rdd] def adamLoad[T](filePath: String, predicate: Option[FilterPredicate] = None, projection: Option[Schema] = None)(implicit ev1: T => SpecificRecord, ev2: Manifest[T]): RDD[T] = {
+  def adamLoad[T](filePath: String, predicate: Option[FilterPredicate] = None, projection: Option[Schema] = None)(implicit ev1: T => SpecificRecord, ev2: Manifest[T]): RDD[T] = {
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: Howabouts we rename this function from `adamLoad` to `loadParquet`?

**Comment by fnothaft** [comment]
Looks good! I think we made it package private under the assumption that you'd use `loadParquet*` where `*` = one of `Alignments`, `Features`, etc.

**Comment by fnothaft** [comment]
Sorry, accidentally closed this!

**Comment by fnothaft** [comment]
Sorry, accidentally closed this!

**Comment by ryan-williams** [comment]
@fnothaft any idea what's causing these build failures? The last two failed builds here both had errors like:

```
2015-05-31 08:09:05 ERROR MetricsSystem:96 - Sink class org.apache.spark.metrics.sink.MetricsServlet cannot be instantialized
java.lang.reflect.InvocationTargetException
```

This last time it was just scala 2.10 x hadoop 1.0.4, whereas previously it was all 2.10 builds

**Comment by ryan-williams** [comment]
rebased and squashed

**Comment by fnothaft** [comment]
Merged! Thanks @ryan-williams!

**Pull request merged**